tests/core/components/test_system_simulation.py

```diff
@@ -0,0 +1,78 @@
+from typing import Any, Iterable
+
+import pytest
+from immutables import Map
+from mock import AsyncMock, Mock, patch
+
+from tickit.core.components.system_simulation import SystemSimulation
+from tickit.core.state_interfaces.internal import (
+    InternalStateConsumer,
+    InternalStateProducer,
+)
```

Comment:
Could we eliminate our dependence on `InternalStateConsumer` & `InternalStateProducer` by using mocks (free or using the protocol for spec)?

Reply:
We can indeed. Doesn't affect whether the tests pass.